#!/bin/env python

import sys
from subprocess import check_call as cmd
from subprocess import check_output as cmd_out
from subprocess import CalledProcessError

out = cmd_out(['git', 'ls-remote', 'origin'])
refs = [line.split("\t") for line in out.strip().split("\n")]

branches = {}
for commit, ref in refs:
	if '/changes/' not in ref: continue
	change_id = ref.split("/")[3] # refs/changes/??/CHANGE_ID/PATCH_NO
	branches[change_id] = ref

for branch, ref in branches.items():
	cmd(['git', 'fetch', 'origin', ref])
	branch = 'change-%s' % branch
	try:
		cmd(['git', 'branch', '-D', branch])
	except CalledProcessError:
		pass
	cmd(['git', 'branch', branch, 'FETCH_HEAD'])
